From 77ccae5bdb10be656c9f0c88ae82847b9312c0c4 Mon Sep 17 00:00:00 2001
From: Hannes Schueller <hannes@yllr.net>
Date: Fri, 26 Dec 2014 11:49:31 +0100
Subject: [PATCH 08/20] default settings: - "s" as shortcut for Startpage
 search - DuckDuckGo as default search engine - Cookie policy: accept no third
 party cookies

---
 config.h        | 7 ++++---
 vimprobable2.1  | 2 ++
 vimprobablerc.5 | 4 ++--
 3 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/config.h b/config.h
index e753824..276fc9f 100644
--- a/config.h
+++ b/config.h
@@ -84,8 +84,8 @@ static URIHandler uri_handlers[] = {
 #define             ENABLE_COOKIE_SUPPORT
 #define             COOKIES_STORAGE_FILENAME    "%s/vimprobable/cookies", client.config.config_base
 #define             COOKIES_STORAGE_READONLY    FALSE   /* if TRUE new cookies will be lost if you quit */
-SoupCookieJarAcceptPolicy CookiePolicy = SOUP_COOKIE_JAR_ACCEPT_ALWAYS; /* by default, accept all cookies */
-SoupCookieJarAcceptPolicy CookiePolicyLastOn = SOUP_COOKIE_JAR_ACCEPT_ALWAYS; /* tracking variable for private mode */
+SoupCookieJarAcceptPolicy CookiePolicy = SOUP_COOKIE_JAR_ACCEPT_NO_THIRD_PARTY; /* by default, accept all cookies, but third party */
+SoupCookieJarAcceptPolicy CookiePolicyLastOn = SOUP_COOKIE_JAR_ACCEPT_NO_THIRD_PARTY; /* tracking variable for private mode */
 
 
 /* font size */
@@ -118,13 +118,14 @@ gboolean complete_case_sensitive        = TRUE;
 /* search engines */
 static Searchengine searchengines[] = {
     { "i",          "http://ixquick.com/do/metasearch.pl?query=%s" },
+    { "s",          "http://startpage.com/do/metasearch.pl?query=%s" },
     { "w",          "https://secure.wikimedia.org/wikipedia/en/w/index.php?title=Special%%3ASearch&search=%s&go=Go" },
     { "wd",         "https://secure.wikimedia.org/wikipedia/de/w/index.php?title=Special%%3ASearch&search=%s&go=Go" },
     { "d",          "https://duckduckgo.com/?q=%s&t=vimprobable" },
     { "dd",         "https://duckduckgo.com/html/?q=%s&t=vimprobable" },
 };
 
-static char defaultsearch[MAX_SETTING_SIZE] = "i";
+static char defaultsearch[MAX_SETTING_SIZE] = "d";
 
 /* command mapping */
 Command commands[COMMANDSIZE] = {
diff --git a/vimprobable2.1 b/vimprobable2.1
index 6ece993..8dcea29 100644
--- a/vimprobable2.1
+++ b/vimprobable2.1
@@ -329,6 +329,8 @@ The following search engines (and shortcuts) are already defined in config.h:
 
 .IP i
 http://ixquick.com/do/metasearch.pl?query=%s
+.IP s
+http://startpage.com/do/metasearch.pl?query=%s
 .IP w
 https://secure.wikimedia.org/wikipedia/en/w/index.php?title=Special%%3ASearch&search=%s&go=Go
 .IP wd
diff --git a/vimprobablerc.5 b/vimprobablerc.5
index 414cd99..4795f0f 100644
--- a/vimprobablerc.5
+++ b/vimprobablerc.5
@@ -1,7 +1,7 @@
 .\" Process this file with
 .\" groff -man -Tascii vimprobablerc.5
 .\"
-.TH VIMPROBABLERC 5 "JANUARY 2014" "Linux User Manuals"
+.TH VIMPROBABLERC 5 "DECEMBER 2014" "Linux User Manuals"
 .SH NAME
 vimprobablerc \- The configuration file for Vimprobable
 .SH DESCRIPTION
@@ -68,7 +68,7 @@ Case sensitive (true) or insensitive tab completion.
 
 .IP cookies=[true|false|no_third]
 Cookie policy: accept all cookies, no cookies or all but third party
-cookies.
+cookies (default).
 
 .IP localstorage=[true|false]
 Enable or disable HTML5 local storage (default: false)
-- 
2.1.4

